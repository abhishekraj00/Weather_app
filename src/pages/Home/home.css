.card_main{
    display: flex;
    flex-wrap: wrap;

    flex-direction: row;
    gap:1rem;

    justify-content: center;
    align-items: center;
}

.cardPice{
margin: 1rem;
padding: 1rem;
}