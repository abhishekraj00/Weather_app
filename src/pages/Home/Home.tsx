import React, { useEffect, useState } from "react";
import CapitalWeather from "../../components/CapitalWeather";
import { getApiData } from "../../utils/Api";
import "./home.css";

interface ResultItem {
  name: {
    common: string;
  };
  capital: string[];
  latlng: number[];
  flags: {
    png: string;
    alt: string;
  };
  population: number;
}

function Home() {
  const [inputData, setInput] = useState<string>("");

  const [callApi, setCall] = useState(false);
  const [countryData, setCountry] = useState<ResultItem[]>([]);

  useEffect(() => {
    if (!inputData) {
      setCountry([]);
    }
  }, [inputData]);

  useEffect(() => {
    const link = `https://restcountries.com/v3.1/name/${inputData}`;
    inputData &&
      getApiData(link).then((res) => {
        setCountry(res.data.slice(0, 4));
      });
  }, [callApi]);

  const handelSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    setCall(!callApi);
  };

  return (
    <>
      <div className="App">
        <h1 className="heading">Weather App</h1>

        {/* From elements */}
        <form onSubmit={handelSubmit}>
          <input
            className="inputBox"
            type="text"
            value={inputData}
            placeholder="Enter Country"
            onChange={(e) => setInput(e.target.value)}
          />
          {inputData && (
            <button className="btn btn-dark inputButton" type="submit">
              Submit
            </button>
          )}
        </form>

        <div className="card_main">
          {countryData &&
            countryData.map((item: ResultItem) => {
              return (
                <div key={item.population} className="card cardPice">
                  <h3>{item.name.common}</h3>
                  <div>
                    <img
                      className="card m-2"
                      src={item.flags.png}
                      alt={item.flags.png}
                      width="120px"
                      height="50px"
                    />
                  </div>
                  <div>
                    <b>Capital</b>: {item.capital[0]}
                  </div>
                  <div>
                    <b>population</b>: {item.population}
                  </div>
                  <div>
                    <b>latitude</b>: {item.latlng[0]}
                  </div>
                  <div>
                    <b>Longitude</b>: {item.latlng[1]}
                  </div>
                  <CapitalWeather capital={item.capital[0]} />
                </div>
              );
            })}
        </div>
      </div>
    </>
  );
}

export default Home;
