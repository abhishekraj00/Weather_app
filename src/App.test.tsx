import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "./pages/Home/Home";
// import Link from '../Link';

// describe("Country List Data", () => {
//   it("should render country list when api call", async () => {
//     render(<Home />);
//     await waitFor(() => {
//       screen.getByText("india");
//     });
//   });
// });

// create snapshot test for each component

describe("Country List Data", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
