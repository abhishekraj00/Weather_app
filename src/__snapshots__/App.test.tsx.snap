// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country List Data renders correctly 1`] = `
<div
  className="App"
>
  <h1
    className="heading"
  >
    Weather App
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inputBox"
      onChange={[Function]}
      placeholder="Enter Country"
      type="text"
      value=""
    />
  </form>
  <div
    className="card_main"
  />
</div>
`;
